<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.29">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Notes Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="notes_files/libs/clipboard/clipboard.min.js"></script>
<script src="notes_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="notes_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="notes_files/libs/quarto-html/popper.min.js"></script>
<script src="notes_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="notes_files/libs/quarto-html/anchor.min.js"></script>
<link href="notes_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="notes_files/libs/quarto-html/quarto-syntax-highlighting-0815c480559380816a4d1ea211a47e91.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="notes_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="notes_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="notes_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Notes Analysis</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div id="89ec534f" class="cell" data-execution_count="2">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">"./reaction_sentences.parquet"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>df_sentences <span class="op">=</span> pd.read_parquet(<span class="st">"./sentences.parquet"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Reaction Sentences dataframe</p>
<div id="f5c1e522" class="cell" data-execution_count="3">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>df.head(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="18">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">original_notes</th>
<th data-quarto-table-cell-role="th">sentences</th>
<th data-quarto-table-cell-role="th">num_sentences</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>[The following mutations in MCCC1 have been id...</td>
<td>8</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>The following ALK mutants are resistant to inh...</td>
<td>[The following ALK mutants are resistant to in...</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Beta-1,3-galactosyltransferase 6 (B3GALT6) tra...</td>
<td>[Beta-1,3-galactosyltransferase 6 (B3GALT6) tr...</td>
<td>3</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Xylosyltransferases (XYLT1, XYLT2) catalyse th...</td>
<td>[Xylosyltransferases (XYLT1, XYLT2) catalyse t...</td>
<td>4</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>UDP-glucuronic acid decarboxylase 1 (UXS1) dec...</td>
<td>[UDP-glucuronic acid decarboxylase 1 (UXS1) de...</td>
<td>5</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>B3GAT1 (Kitagawa et al. 1998) and B3GAT2 (Marc...</td>
<td>[B3GAT1 (Kitagawa et al. 1998) and B3GAT2 (Mar...</td>
<td>4</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>Galactosylgalactosylxylosylprotein 3-beta-gluc...</td>
<td>[Galactosylgalactosylxylosylprotein 3-beta-glu...</td>
<td>4</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>Beta-1,4-galactosyltransferase 7 (B4GALT7) add...</td>
<td>[Beta-1,4-galactosyltransferase 7 (B4GALT7) ad...</td>
<td>2</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>APC is K63 polyubiquitinated in the absence of...</td>
<td>[APC is K63 polyubiquitinated in the absence o...</td>
<td>6</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>The G-protein alpha subunit exchanges GDP for GTP</td>
<td>[The G-protein alpha subunit exchanges GDP for...</td>
<td>1</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="4911efc1" class="cell" data-execution_count="4">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of rows:</span><span class="sc">{</span><span class="bu">len</span>(df)<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of rows:14802</code></pre>
</div>
</div>
<p>Sentences dataframe</p>
<div id="4cf7a3f3" class="cell" data-execution_count="5">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>df_sentences.head(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="20">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">note_index</th>
<th data-quarto-table-cell-role="th">original_notes</th>
<th data-quarto-table-cell-role="th">sentence</th>
<th data-quarto-table-cell-role="th">num_sentences</th>
<th data-quarto-table-cell-role="th">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>They have been shown to have compromised activ...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>R385S: appears to affect protein activity with...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>May act as a dominant negative (Steen et al, 1...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>L437P: abolishes enzyme activity (Baumgartner ...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>S173L: no detectable MCCC activity in expressi...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>H190Y: mutation of a highly conserved residue,...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>0</td>
<td>The following mutations in MCCC1 have been ide...</td>
<td>G352R: mutation of a highly conserved residue,...</td>
<td>8</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>1</td>
<td>The following ALK mutants are resistant to inh...</td>
<td>The following ALK mutants are resistant to inh...</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>2</td>
<td>Beta-1,3-galactosyltransferase 6 (B3GALT6) tra...</td>
<td>Beta-1,3-galactosyltransferase 6 (B3GALT6) tra...</td>
<td>3</td>
<td>1</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="0f29fba7" class="cell" data-execution_count="6">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of rows:</span><span class="sc">{</span><span class="bu">len</span>(df_sentences)<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of rows:53380</code></pre>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="bu">max</span> <span class="op">=</span> <span class="dv">5</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">0</span>,<span class="bu">max</span>):</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  display(Markdown(<span class="ss">f"</span><span class="ch">\n</span><span class="ss"> **Original notes at index </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">:**  </span><span class="ch">\n</span><span class="sc">{</span>df[<span class="st">'original_notes'</span>]<span class="sc">.</span>iloc[idx]<span class="sc">}</span><span class="ss"> </span><span class="ch">\n</span><span class="ss">"</span>))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  sentences <span class="op">=</span> <span class="st">"  </span><span class="ch">\n</span><span class="st">"</span>.join([<span class="ss">f"**-** </span><span class="sc">{</span>s<span class="sc">}</span><span class="ss">"</span> <span class="cf">for</span> s <span class="kw">in</span> df[<span class="st">'sentences'</span>].iloc[idx]])</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  display(Markdown(<span class="ss">f"**Sentences of note at index </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">:**  </span><span class="ch">\n</span><span class="sc">{</span>sentences<span class="sc">}</span><span class="ss">"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<p><strong>Original notes at index 0:</strong><br>
The following mutations in MCCC1 have been identified in patients with MCCC deficiency I. They have been shown to have compromised activity in vitro and are classified as either “likely pathological” or “pathological” in ClinVar despite variable clinical presentations:M325R: appears to abolish the ability of the protein to be biotinylated, possibly affecting protein stability (Gallardo et al, 2001).R385S: appears to affect protein activity without affecting stability. May act as a dominant negative (Steen et al, 1999; Gallardo et al, 2001; Baumgartner et al, 2001; Baumgartner et al, 2004; Fonseca et al, 2016; Dantas et al, 2005; Morscher et al, 2012).L437P: abolishes enzyme activity (Baumgartner et al, 2001)I460M: reduces enzyme activity to D532H: This mutation arises as a result of a nucleotide change causing an alteration of a splice site, and results in a truncated protein (Baumgartner et al, 2001)V694*: This mutation leads to premature truncation and removes the essential biotinylation site (Holzinger et al, 2001)The following mutations in MCCC2 have been identified in patients with MCCC deficiency II, have been shown to have compromised activity in vitro and are classified as either “likely pathological” or “pathological” in ClinVar, despite variable clinical presentations:E99Q: this mutation shows no detectable MCCC activity in expression studies and variable clinical outcome (Baumgartner et al, 2001; Grünert et al, 2012).S173L: no detectable MCCC activity in expression studies (Baumgartner et al, 2001).H190Y: mutation of a highly conserved residue, severely decreased MCCC activity (Dantas et al, 2005)V339M: ~4% residual MCCC activity with severe clinical outcome in two Turkish patients (Baumgartner et al, 2001).G352R: mutation of a highly conserved residue, severely decreased MCCC activity (Dantas et al, 2005; Grünert et al, 2012)A456V: mutation of a highly conserved residue, virtually no detectable MCCC activity (Dantas et al, 2005; Grünert et al, 2012).</p>
<p><strong>Sentences of note at index 0:</strong><br>
<strong>-</strong> The following mutations in MCCC1 have been identified in patients with MCCC deficiency I.<br>
<strong>-</strong> They have been shown to have compromised activity in vitro and are classified as either “likely pathological” or “pathological” in ClinVar despite variable clinical presentations:M325R: appears to abolish the ability of the protein to be biotinylated, possibly affecting protein stability (Gallardo et al, 2001).<br>
<strong>-</strong> R385S: appears to affect protein activity without affecting stability.<br>
<strong>-</strong> May act as a dominant negative (Steen et al, 1999; Gallardo et al, 2001; Baumgartner et al, 2001; Baumgartner et al, 2004; Fonseca et al, 2016; Dantas et al, 2005; Morscher et al, 2012).<br>
<strong>-</strong> L437P: abolishes enzyme activity (Baumgartner et al, 2001)I460M: reduces enzyme activity to D532H: This mutation arises as a result of a nucleotide change causing an alteration of a splice site, and results in a truncated protein (Baumgartner et al, 2001)V694*: This mutation leads to premature truncation and removes the essential biotinylation site (Holzinger et al, 2001)The following mutations in MCCC2 have been identified in patients with MCCC deficiency II, have been shown to have compromised activity in vitro and are classified as either “likely pathological” or “pathological” in ClinVar, despite variable clinical presentations:E99Q: this mutation shows no detectable MCCC activity in expression studies and variable clinical outcome (Baumgartner et al, 2001; Grünert et al, 2012).<br>
<strong>-</strong> S173L: no detectable MCCC activity in expression studies (Baumgartner et al, 2001).<br>
<strong>-</strong> H190Y: mutation of a highly conserved residue, severely decreased MCCC activity (Dantas et al, 2005)V339M: ~4% residual MCCC activity with severe clinical outcome in two Turkish patients (Baumgartner et al, 2001).<br>
<strong>-</strong> G352R: mutation of a highly conserved residue, severely decreased MCCC activity (Dantas et al, 2005; Grünert et al, 2012)A456V: mutation of a highly conserved residue, virtually no detectable MCCC activity (Dantas et al, 2005; Grünert et al, 2012).</p>
<p><strong>Original notes at index 1:</strong><br>
The following ALK mutants are resistant to inhibition by ASP-3026:ALK T1551M: Amin et al 2016ALK I1171S: Amin et al 2016ALK I1171T: Katayama et al, 2014ALK F1174L: Amin et al, 2016 (moderate)ALKR 1192P: Amin et al, 2016ALK G1269A: Amin et al, 2016 (moderate)</p>
<p><strong>Sentences of note at index 1:</strong><br>
<strong>-</strong> The following ALK mutants are resistant to inhibition by ASP-3026:ALK T1551M: Amin et al 2016ALK I1171S: Amin et al 2016ALK I1171T: Katayama et al, 2014ALK F1174L: Amin et al, 2016 (moderate)ALKR 1192P: Amin et al, 2016ALK G1269A: Amin et al, 2016 (moderate)</p>
<p><strong>Original notes at index 2:</strong><br>
Beta-1,3-galactosyltransferase 6 (B3GALT6) transfers a second galactose to the tetrasaccharide linker. Although it can act on substrates with a terminal beta-linked galactose residue, it prefers the sequence galactose-beta-1,4-xylose (Bai et al.&nbsp;2001). B3GALT6 requires manganese as a cofactor (Zhou et al.&nbsp;1999).</p>
<p><strong>Sentences of note at index 2:</strong><br>
<strong>-</strong> Beta-1,3-galactosyltransferase 6 (B3GALT6) transfers a second galactose to the tetrasaccharide linker.<br>
<strong>-</strong> Although it can act on substrates with a terminal beta-linked galactose residue, it prefers the sequence galactose-beta-1,4-xylose (Bai et al.&nbsp;2001).<br>
<strong>-</strong> B3GALT6 requires manganese as a cofactor (Zhou et al.&nbsp;1999).</p>
<p><strong>Original notes at index 3:</strong><br>
Xylosyltransferases (XYLT1, XYLT2) catalyse the initial step in the tetrasaccharide linkage required for glycosaminoglycan biosynthesis. This reaction can take place in the Golgi apparatus and endoplasmic reticulum (not shown here). XYLTs mediates the transfer of xylose from the active nucleotide sugar UDP-xylose to specific serine hydroxy groups in the core protein. A C-terminal DxD motif on the enzyme is thought to be critical for activity (Muller et al.&nbsp;2005, Goetting et al.&nbsp;2004, Voglmeir et al.&nbsp;2007).</p>
<p><strong>Sentences of note at index 3:</strong><br>
<strong>-</strong> Xylosyltransferases (XYLT1, XYLT2) catalyse the initial step in the tetrasaccharide linkage required for glycosaminoglycan biosynthesis.<br>
<strong>-</strong> This reaction can take place in the Golgi apparatus and endoplasmic reticulum (not shown here).<br>
<strong>-</strong> XYLTs mediates the transfer of xylose from the active nucleotide sugar UDP-xylose to specific serine hydroxy groups in the core protein.<br>
<strong>-</strong> A C-terminal DxD motif on the enzyme is thought to be critical for activity (Muller et al.&nbsp;2005, Goetting et al.&nbsp;2004, Voglmeir et al.&nbsp;2007).</p>
<p><strong>Original notes at index 4:</strong><br>
UDP-glucuronic acid decarboxylase 1 (UXS1) decarboxylates UDP-alpha-D-glucuronate (UDP-GlcA) to form UDP-alpha-D-xylose (UDP-Xyl). The reaction is localized to the Golgi apparatus (inferred from studies of the homologous rat enzyme (Moriarity et al.&nbsp;2002). UXS1 is associated with NAD+ (Eixelberger et al.&nbsp;2012; Walsh et al.&nbsp;2015 ). The active form of the enzyme is a tetramer of four UXS1:NAD+ subunits (Polizzi et al 2013). UDP-xylose is then used to initiate glycosaminoglycan biosynthesis on the core protein of proteoglycans.</p>
<p><strong>Sentences of note at index 4:</strong><br>
<strong>-</strong> UDP-glucuronic acid decarboxylase 1 (UXS1) decarboxylates UDP-alpha-D-glucuronate (UDP-GlcA) to form UDP-alpha-D-xylose (UDP-Xyl).<br>
<strong>-</strong> The reaction is localized to the Golgi apparatus (inferred from studies of the homologous rat enzyme (Moriarity et al.&nbsp;2002).<br>
<strong>-</strong> UXS1 is associated with NAD+ (Eixelberger et al.&nbsp;2012; Walsh et al.&nbsp;2015 ).<br>
<strong>-</strong> The active form of the enzyme is a tetramer of four UXS1:NAD+ subunits (Polizzi et al 2013).<br>
<strong>-</strong> UDP-xylose is then used to initiate glycosaminoglycan biosynthesis on the core protein of proteoglycans.</p>
<div id="a61185e7" class="cell" data-execution_count="8">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>df.sort_values(by<span class="op">=</span><span class="st">"num_sentences"</span>, ascending<span class="op">=</span><span class="va">False</span>).head(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="23">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">original_notes</th>
<th data-quarto-table-cell-role="th">sentences</th>
<th data-quarto-table-cell-role="th">num_sentences</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">6012</td>
<td>During differentiation of stratum granulosum k...</td>
<td>[During differentiation of stratum granulosum ...</td>
<td>128</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5988</td>
<td>Trunk bipotent pancreatic progenitor cells (bi...</td>
<td>[Trunk bipotent pancreatic progenitor cells (b...</td>
<td>105</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5986</td>
<td>Pancreatic pro-acinar cells mature into pancre...</td>
<td>[Pancreatic pro-acinar cells mature into pancr...</td>
<td>91</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">6016</td>
<td>The formation of foregut endoderm cells from d...</td>
<td>[The formation of foregut endoderm cells from ...</td>
<td>67</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6020</td>
<td>The formation of foregut endoderm cells from d...</td>
<td>[The formation of foregut endoderm cells from ...</td>
<td>61</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7151</td>
<td>Transcriptional coactivators recruited to the ...</td>
<td>[Transcriptional coactivators recruited to the...</td>
<td>59</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5593</td>
<td>Optineurin (OPTN, also called FIP2 and NRP), i...</td>
<td>[Optineurin (OPTN, also called FIP2 and NRP), ...</td>
<td>42</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">10976</td>
<td>Angiotensin-converting enzyme (ACE) is a centr...</td>
<td>[Angiotensin-converting enzyme (ACE) is a cent...</td>
<td>42</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">423</td>
<td>β-adrenergic receptor antagonists ("β-blockers...</td>
<td>[β-adrenergic receptor antagonists ("β-blocker...</td>
<td>41</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">11857</td>
<td>Acetylcholinesterase (ACHE) is involved in the...</td>
<td>[Acetylcholinesterase (ACHE) is involved in th...</td>
<td>41</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>display(Markdown(<span class="ss">f"**Note with max value of sentences**</span><span class="ch">\n</span><span class="sc">{</span>df[<span class="st">'original_notes'</span>]<span class="sc">.</span>iloc[df[<span class="st">'num_sentences'</span>].idxmax()]<span class="sc">}</span><span class="ch">\n</span><span class="ss">"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<p><strong>Note with max value of sentences</strong> During differentiation of stratum granulosum keratinocytes into corneocytes, lytic enzymes are activated by the influx of Ca2+ and all metabolic activity ceases. Corneocytes, in the form of flattened squames, represent cellular skeletons filled with keratin filaments and sealed together by lipids. Corneocytes form an impermeable and insoluble barrier that protects against the invasion of microorganisms and loss of bodily fluids (reviewed by Fuchs 1990).Markers of granular keratinocytes are summarized in the “Table of granular keratinocytes in interfollicular epidermis”. Some of the markers, in particular AQP3, LGLAS7, and FABP5, are described in more detail in the sections “Keratinocyte stem cell differentiates into transit amplifying cell in the basal layer of interfollicular epidermis” and “Keratinocyte of spinosum layer differentiates into keratinocyte of granulosum layer in interfollicular epidermis”.CALML5 (Calmodulin-like protein 5, also known as Calmodulin-like skin protein or CLSP) is expressed in keratinocytes of stratum granulosum and lower layers of stratum corneum and is a marker of late keratinocyte differentiation (Mehul et al.&nbsp;2001). CALML5 is proteolytically degraded in upper layers of normal but not psoriatic stratum corneum, which is prevented by Ca2+ (Mehul et al.&nbsp;2006). CALML5 is implicated in regulation of function of Ca2+-dependent proteins involved in skin barrier formation (Donovan et al.&nbsp;2015). By RNA sequencing, CALML5 is the most enriched gene in differentiating outer epidermis. CALML5 gene transcription is upregulated by ZNF750 transcription factor and CALML5 mRNA is stabilized by noncoding RNA TINCR; CALML5 knockdown interferes with terminal differentiation of keratinocytes, formation of keratohyalin granules, and the skin barrier function. SFN is a binding partner of CALML5 (Sun et al.&nbsp;2015).Table of markers of granular keratinocytes in interfollicular epidermis. Please note that keratinocytes in CellMarker database and PanglaoDB correspond to granular keratinocytes in Reactome.Marker (protein/RNA)Literature ReferenceCellMarker database – RNA/Protein (Hu et al.&nbsp;2022)PanglaoDB – RNA (Franzén et al.&nbsp;2019)AQP3 (protein)Sugiyama et al.&nbsp;2001Ma et al.&nbsp;2002 Sougrat et al.&nbsp;2002NoYesCALML5 (protein)Mehul et al.&nbsp;2001Mehul et al.&nbsp;2006Donovan et al.&nbsp;2015Sun et al.&nbsp;2015NoYesDMKN (protein)Matsui et al.&nbsp;2004Higashi et al.&nbsp;2012Leclerc et al.&nbsp;2014NoYesFAPB5 (protein)Let et al.&nbsp;1998NoFLG (protein)Smoller et al.&nbsp;1990Steven et al.&nbsp;1990Juhlin et al.&nbsp;1992Girbal Neuhauser et al.&nbsp;1997Yoneda et al.&nbsp;2012Sakabe et al.&nbsp;2013Wong et al.&nbsp;2015reviewed in McLean 2016reviewed in Ishida Yamamoto et al.&nbsp;2018NoYesGJB3 (protein)Butterweck et al.&nbsp;1994Di et al.&nbsp;2001NoYesHOPX (protein)Obarzanek Fojt et al.&nbsp;2011NoNoIVL (protein)Bernard et al.&nbsp;1985Smoller et al.&nbsp;1990van Duijnhoven et al.&nbsp;1992Ishida Yamamoto et al.&nbsp;1996Crish et al.&nbsp;2002Chen et al.&nbsp;2013NoYes&gt;KLK5 (protein)Ishida Yamamoto et al.&nbsp;2005Sakabe et al.&nbsp;2013Miyai et al.&nbsp;2014NoNoKLK7 (protein)Ishida Yamamoto et al.&nbsp;2005Miyai et al.&nbsp;2014NoNoKPRP (protein)Lee et al.&nbsp;2005Suga et al.&nbsp;2019Jeriha et al.&nbsp;2020NoYesKRT1 (protein)reviewed in Fuchs 1990Totsuka et al.&nbsp;2017NoYesKRT10 (protein)Yoneda et al.&nbsp;2012Totsuka et al.&nbsp;2017NoYesKRTDAP (protein)Matsui et al.&nbsp;2004Tsuchida et al.&nbsp;2004NoNoLGALS7 (protein)Magnaldo et al.&nbsp;1995Umayahara et al.&nbsp;2020NoYesLORICRIN (protein, RNA)Steven et al.&nbsp;1990Juhlin et al.&nbsp;1992Magnaldo et al.&nbsp;1992Ishida Yamamoto et al.&nbsp;1996reviewed in Ishida Yamamoto et al.&nbsp;1998Yoneda et al.&nbsp;2012reviewed in Ishida Yamamoto et al.&nbsp;2018NoNoPRSS3 (protein)Nakanishi et al.&nbsp;2010Miyai et al.&nbsp;2014NoNoSBSN (protein)Matsui et al.&nbsp;2004Bazzi et al.&nbsp;2007Aoshima et al.&nbsp;2019NoYesSPINK5, also known as LEKTI (protein)Bitoun et al.&nbsp;2003Ishida Yamamoto et al.&nbsp;2005Galliano et al.&nbsp;2005reviewed in Ishida Yamamoto et al.&nbsp;2018NoNoSPRR2A (protein)Hohl et al.&nbsp;1995Sark et al.&nbsp;1998Cabral et al.&nbsp;2003YesNoDMKN (dermokine)-endcoding gene is part of the keratinocyte related locus on human chromosome 19q13.1 that also includes SBSN and KRTDAP genes. Coordinated transcription of these genes is detectable in stratum spinosum, but proteins are the most prominent in the granular layer keratinocytes, from which they are secreted. DMKN has several splicing isoforms, of which alpha, beta, and gamma are known to be expressed in the epidermis (Matsui et al.&nbsp;2004). DMKN beta splicing isoform was shown to inhibit activating phosphorylation of MAPK1/MAPK3, which depends on the interaction of DMKN with GRP78 (Higashi et al.&nbsp;2012). The pups of mice deficient in Dmkn beta have smaller keratohyalin granules and their cornified envelopes are more sensitive to mechanical stress (Leclerc et al.&nbsp;2014).In human skin biopsies, FLG (filaggrin) is expressed in all keratinocytes above the stratum spinosum (Smoller et al.&nbsp;1990), including keratinocytes of the granular layer of epidermis (Juhlin et al.&nbsp;1992). In newborn mouse skin, FLG is first detected in large, irregular shaped keratohyalin granules in the keratinocytes of stratum granulosum, and then distributed throughout the cytoplasm of the corneocytes of the innermost layer of stratum corneum (Steven et al.&nbsp;1990). Anti-(pro)filaggrin autoantibodies derived from patients with rheumatoid arthritis produce granular staining of stratum granulosum (Girbal Neuhauser et al.&nbsp;1997). FLG is detectable in the cytosol and nucleus of stratum granulosum cells formed by cultured human keratinocytes. As FLG undergoes proteolytic cleavage upon terminal differentiation of granular keratinocytes, FLG cleavage products are detectable (Yoneda et al.&nbsp;2012). Profilaggrin is expressed in the stratum granulosum of human epidermis by immunochemistry (Sakabe et al.&nbsp;2013). Hypoxia leads to HIF1A- and EPAS1 (HIF2A)-mediated stimulation of FLG transcription in a Ca2+-independent and cell confluency-independent manner, and stimulates expression of SPINK5, an inhibitor of epidermal serine proteases that regulates conversion of profilaggrin into monomeric filaggrin. Three putative hypoxia response elements (HREs) are found in the promoter region of the human FLG gene, of which two HREs are functional. One of the two functional HREs is evolutionarily conserved. Direct binding of HIF1A/EPAS1 to these HREs has not been demonstrated. Simultaneous HIF1A and EPAS1 deficiency inhibits terminal differentiation of keratinocytes and formation of stratum granulosum and cornified envelope. ARNT Aryl hydrocarbon receptor nuclear translocator involved in ceramide synthesis also positively regulates FLG expression (Wong et al.&nbsp;2015). Profilaggrin, a protein of &gt;400 kDa that is the major protein constituent of keratohyalin granules in keratinocytes of stratum granulosum, consists of 10 to 12 near identical repeats of FLG monomers connected by short cleavable linkers. When granular keratinocytes commit to differentiation into corneocytes, profilaggrin is cleaved and 37 kDa FLG monomers are release. FLG monomers bind to and condense keratin cytoskeleton, facilitating cellular compression. In the stratum corneum, FLG monomers are further broken down to form natural moisturizing factor - NMF - a pool of hygroscopic amino acids and derivatives. Loss-of-function mutations in the FLG gene, present in ~10% of human population, are the cause of ichthyosis vulgaris, dry flaky skin phenotype. Ichthyosis vulgaris is mild in heterozygous carriers, and the phenotype severity also depends on environmental and lifestyle factors, being more pronounced in cold, dry climates, and ameliorated by the use of skin moisturizers. A naturally occurring Flg mutation in mice results in the “flaky tail” phenotype. Loss-of-function FLG mutations as well as subtle size variations in the FLG gene predispose carriers to atopic conditions such as eczema and allergy, by compromising the skin barrier and allowing allergens to cross the skin epidermis, triggering Th-2 allergic immunity (reviewed in McLean 2016). Profilaggrin, synthesized in granular keratinocytes, is proteolyzed into FLG monomers in cornified keratinocytes, concurrent with disappearance of nucleus and cell organelles and FLG mediated aggregation of keratin filaments into a tight matrix (reviewed in Ishida Yamamoto et al.&nbsp;2018).GJB3 (Gap junction beta 3 protein, also known as Connexin 31 or CX31), is expressed in stratum granulosum of mouse epidermis (Butterweck et al.&nbsp;1994). In normal human epidermis, GJB3 is predominantly expressed in stratum granulosum, with a punctate staining at the plasma membrane, consistent with its involvement in formation of gap junctions. Mutations in GJB3 gene are associated with erythrokeratoderma variabilis (Di et al.&nbsp;2001).HOPX is an atypical homeodomain protein unable to bind to DNA, expressed in keratinocytes of stratum granulosum of human epidermis, and induced during Ca2+-mediated differentiation of cultured primary human keratinocytes. Overexpression of HOPX leads to upregulation of FLG and LORICRIN, while HOPX knockdown interferes with Ca2+-mediated induction of FLG and LORICRIN expression. Flg and Loricrin mRNA levels are downregulated in the epidermis of Hopx knockout mice (Obarzanek Fojt et al.&nbsp;2011).IVL is detected in stratum granulosum in normal epidermis and all layers except the basal layer in psoriatic epidermis (Bernard et al.&nbsp;1985). In human skin biopsies, involucrin is expressed in all keratinocytes above the stratum spinosum (Smoller et al.&nbsp;1990). Human IVL is expressed in keratinocytes of stratum granulosum, with expression increasing as the cells reach confluence (van Duijnhoven et al.&nbsp;1992). In the keratinocytes of stratum granulosum, IVL is detected along the plasma membrane. In the corneocytes, involucrin labeling is reduced, but it is very intense in the psoriatic epidermis. Formation of the cornified envelope likely first involves the deposition of IVL, followed by incorporation of LORICRIN. In psoriatic epidermis, IVL deposition at the cornified envelope occurs precociously, without further maturation of the cornified envelope (Ishida Yamamoto et al.&nbsp;1996). The distal regulatory region of the human IVL gene promoter, located at 2 2.5 kb upstream of the transcription start site, is needed to recapitulate the normal pattern of IVL expression in the epidermis of transgenic mice. The distal regulatory region contains two distinct, spatially separate regulatory elements that likely bind AP1 transcription factors (Crish et al.&nbsp;2002). In the normal skin, IVL is detectable in the stratum granulosum and upper layers of stratum spinosum. IVL expression may be upregulated by IL13, IL17A, EDN1, TNF, IFNG, and MAPK1/MAPK3 and GSK3B signaling (Chen et al.&nbsp;2013).In normal epidermis, KLK5 (Kallikrein-5) and KLK7 (Kallikrein-7) are found in lamellar granules in stratum granulosum, localized separately from SPINK5, which is also found in lamellar granules (Ishida Yamamoto et al.&nbsp;2005). KLK5 is expressed in stratum granulosum of human epidermis by immunochemistry. KLK5 co localizes with profilaggrin and is responsible for profilaggrin cleavage at the linker domain and release of FLG monomers (Sakabe et al.&nbsp;2013). KLK5 and KLK7 co-localize with PRSS3 in the upper layers of stratum granulosum and can be activated by PRSS3 mediated proteolytic cleavage. KLK5 is thought to be the initiator of the kallikrein proteolytic cascade that leads to activation of kallikrein-related peptidases - KLKs - and degradation of corneodesmosomes, leading to desquamation - shedding of the outermost layer of corneocytes (Miyai et al.&nbsp;2014).Expression of KPRP gene, encoding human keratinocyte proline rich protein, is induced by Ca2+ in cultured normal human epidermal keratinocytes. KPRP is predominantly expressed in upper layers of stratum granulosum. In psoriatic epidermis, KPRP levels are increased compared to normal epidermis (Lee et al.&nbsp;2005). KPRP is expressed in upper layers of stratum granulosum of human epidermis and is significantly decreased in atopic dermatitis compared to normal skin. KPRP co localizes with LORICRIN and is mainly detected in the cytoskeleton fraction of human keratinocytes (Suga et al.&nbsp;2019). In an in vitro model of Ca2+ induced differentiation of human keratinocytes, KPRP is a marker of stratum granulosum, consistent with its in vivo expression pattern (Jeriha et al.&nbsp;2020).While keratinocytes in stratum granulosum stop transcribing keratin genes, keratin proteins, including KRT1, are still detectable (Totsuka et al.&nbsp;2017, reviewed in Fuchs 1990).KRT10 protein is detectable in stratum granulosum keratinocytes (Totsuka et al.&nbsp;2017). In stratum granulosum formed by cultured human keratinocytes, KRT10 protein is detectable in the cytosol and nucleus, and co-localizes with the N-terminal cleavage product of profilaggrin (Yoneda et al.&nbsp;2012).KRTDAP gene, encoding keratinocyte differentiation-associated protein (also known as KDAP), is part of the keratinocyte related locus on human chromosome 19q13.1 that also includes SBSN and DKMN genes. Coordinated transcription of these genes is detectable in stratum spinosum, but proteins are the most prominent in the granular layer keratinocytes, from which they are secreted (Matsui et al.&nbsp;2004). In normal human skin, KRTDAP protein is found exclusively in lamellar granules of keratinocytes in stratum granulosum and in the intercellular space in stratum corneum. In psoriatic skin, KRTDAP is expressed more widely and detected in all suprabasal keratinocytes (Tsuchida et al.&nbsp;2004).In newborn mouse skin, LORICRIN is first detected in small, round, keratohyalin granules in keratinocytes of stratum granulosum, and then at the periphery of corneocytes throughout the stratum corneum (Steven et al.&nbsp;1990). LORICRIN is expressed in keratinocytes of granular layer of human epidermis. In skin disorders characterized by incomplete differentiation, e.g.&nbsp;psoriasis, LORICRIN is not detected but FLG is still expressed (Juhlin et al.&nbsp;1992). LORICRIN mRNA and protein are present in stratum granulosum and both transcript and protein are diminished upon treatment of human epidermis culture with retinoic acid (Magnaldo et al.&nbsp;1992). LORICRIN is distributed diffusely in keratinocytes of stratum granulosum, occasionally in granular aggregates in the cytoplasm and nucleus, and concentrated on the desmosomal attachment plaques where it co-localizes with desmoglein. LORICRIN is diminished in psoriatic epidermis (Ishida Yamamoto et al.&nbsp;1996). LORICRIN, diffusely distributed in granular keratinocytes, is associated with the cornified envelope in corneocytes, a tough structure beneath the corneocyte plasma membrane. Mutations in LORICRIN gene are implicated in two hereditary skin disorders, Vohwinkel’s syndrome and progressive symmetric erythrokeratoderma (reviewed in Ishida Yamamoto et al.&nbsp;1998). LORICRIN is detectable in the cytosol and nucleus of cultured human keratinocytes in stratum granulosum, where it co-localizes with the N-terminal cleavage product of FLG (Yoneda et al.&nbsp;2012). As granular keratinocytes differentiate into corneocytes, cornified envelope is formed by covalent crosslinking through transglutamination of various proteins, including LORICRIN, which is one of the main constituents of the cornified envelope (reviewed in Ishida Yamamoto et al.&nbsp;2018).PRSS3, also known as mesotrypsin, is expressed only in the granular layer of epidermis. In cultured keratinocytes, PRSS3 mRNA is expressed at the confluent stage and is strongly upregulated after air exposure (Nakanishi et al.&nbsp;2010). PRSS3 is expressed in the upper granular layer of epidermis, where KLKs and SPINK5 are also found. PRSS3 is able to proteolytically activate KLK5 and KLK7, and hydrolyze SPINK5 fragments bound to KLKs, thus contributing to the desquamation process. PRSS3 expression level in epidermis decreases after the age of 60 (Miyai et al.&nbsp;2014).SBSN gene, encoding Suprabasin, is part of the keratinocyte related locus on human chromosome 19q13.1 that also includes KRTDAP and DKMN genes. Coordinated transcription of these genes is detectable in stratum spinosum, but proteins are the most prominent in the granular layer keratinocytes, from which they are secreted (Matsui et al.&nbsp;2004). The cluster of Sbsn, Krtdap, and Dkmn genes is conserved on mouse chromosome 7, and these genes are coordinately expressed during epidermal differentiation (Bazzi et al.&nbsp;2007). SBSN deficient human skin three dimensional model has compact stratum corneum, immature stratum granulosum, and increased keratinocyte apoptosis (Aoshima et al.&nbsp;2019).SPINK5, also known as serine protease inhibitor LEKTI, is strongly expressed in keratinocytes of stratum granulosum and uppermost layers of keratinocytes of stratum spinosum. Two isoforms of SPINK5, of 145 kDa and 125 kDa, are detectable in keratinocytes. Both isoforms are glycosylated ad processed by a FURIN-like enzyme in the post-endoplasmic reticulum compartment, yielding fragments of 42 kDa, 65 kDa, and 68 kDa. SPINK5 genetic defects lead to Netherthon syndrome, a severe autosomal recessive ichtyosis (Bitoun et al.&nbsp;2003). In normal epidermis, SPINK5 is found in lamellar granules, membrane-bound organelles of stratum granulosum and uppermost layers of stratum spinosum, that fuse with the plasma membrane and secrete their contents into the extracellular space. SPINK5 localizes separately from KLK5 and KLK7, which are also found in lamellar granules, and is likely secreted before KLK5 and KLK7 (Ishida Yamamoto et al.&nbsp;2005). SPINK5 is expressed in keratinocytes of stratum granulosum of both human and mouse epidermis (Galliano et al.&nbsp;2005). SPINK5 cleavage fragments function as inhibitors of KLK5 and KLK7 proteases, preventing premature degradation of corneodesmosomes by KLK5/KLK7, and thus maintaining integrity and cohesion of the stratum corneum (reviewed in Ishida Yamamoto et al.&nbsp;2018).SPRR2A gene is part of a gene cluster of 1.5 Mb on human chromosomal band 1q21 that includes several other small proline-rich protein genes, as well as LORICRIN and IVL genes, which have all likely evolved from a common ancestor. By immunohistochemistry, SPRR2A protein is expressed at moderate levels in stratum granulosum of interfollicular epidermis, and by in situ RNA hybridization SPRR2A mRNA is restricted to the uppermost spinous and granular epidermis layers. Retinoic acid inhibit accumulation of SPRR2A mRNA and protein in keratinocytes. SPRR2A may be a precursor protein of the cornified envelope (Hohl et al.&nbsp;1995). SPRR2A expression is strongly induced by binding of the epithelium-specific transcription factor ELF3, also known as ESE-1, upregulated during Ca2+-induced terminal differentiation of keratinocytes, to the Ets site in the SPRR2A gene promoter (Sark et al.&nbsp;1998). POU2F3 transcription factor isoform Skn-1a cooperates with the transcription factor ELF3 in the induction of SPRR2A gene expression during terminal differentiation of keratinocytes in the epidermis (Cabral et al.&nbsp;2003).Markers of the stratum corneum keratinocytes (corneocytes) are summarized in the “Table of markers of corneocytes in interfollicular epidermis”. Some of the markers, in particular LGLAS7, are described in more detail in the section “Keratinocyte stem cell differentiates into transit amplifying cell in the basal layer of interfollicular epidermis”.Table of markers of corneocytes in interfollicular epidermis. Please note that keratinocytes in CellMarker database and PanglaoDB correspond to corneocytes in Reactome.Marker (protein/RNA)Literature ReferenceCellMarker database – RNA/Protein (Hu et al.&nbsp;2022)PanglaoDB – RNA (Franzén et al.&nbsp;2019)CALML5 (protein)Mehul et al.&nbsp;2001Mehul et al.&nbsp;2006NoYesCASP14Fischer et al.&nbsp;2004Alibardi et al.&nbsp;2004NoNoFLGSimon et al.&nbsp;1996Girbal Neuhauser et al.&nbsp;1997Pendaries et al.&nbsp;2014NoYesKLK5Caubet et al.&nbsp;2004de Veer et al.&nbsp;2017NoNoKLK7Caubet et al.&nbsp;2004de Veer et al.&nbsp;2017NoNoLGALS7Magnaldo et al.&nbsp;1995Umayahara et al.&nbsp;2020NoYesCALML5 is expressed in keratinocytes of lower layers of the stratum corneum. CALML5 is proteolytically degraded in upper layers of normal but not psoriatic stratum corneum, which is prevented by Ca2+ (Mehul et al.&nbsp;2006).CASP14 is highly expressed in differentiated epidermal keratinocytes and is the predominant caspase in the stratum corneum where it is normally present exclusively in its processed form. Both pro-caspase-14 and CASP14 subunits produced by cleavage are present in incompletely matured stratum corneum in psoriasis and seborrheic dermatitis (Fischer et al.&nbsp;2004). CASP14 is first detectable in the nuclei and keratohyalin granules of stratum granulosum keratinocytes, and it is present in the cornified envelope, cytosol, and nucleus of corneocytes (Alibardi et al.&nbsp;2004).FLG partially co-localizes with the cornified envelope in corneocytes (Simon et al.&nbsp;1996). Anti-(pro)filaggrin autoantibodies derived from patients with rheumatoid arthritis produce diffuse staining of stratum corneum (Girbal-Neuhauser et al.&nbsp;1997). FLG is expressed in corneocytes and needed for the maintenance of the corneocyte intracellular matrix.KLK5 (Caubet et al.&nbsp;2004). KLK5 is expressed in stratum corneum and is able to directly cleave all three components of corneodesmosomes, CDSN, DSC1, and DSG1, and to proteolytically activate KLK7; de Veer et al.&nbsp;2017: KLK5 can be isolated from the plantar stratum corneum, and contributes to desquamation).KLK7 is expressed in the stratum corneum and can directly cleave CDSN and DSC1 components of corneodesmosomes, but not DSG1 (Caubet et al.&nbsp;2004). KLK7 can be isolated from the plantar stratum corneum and is the main protease involved in desquamation (de Veer et al.&nbsp;2017).LGALS7 is expressed in all epidermal layers, especially highly in the stratum corneum (Umayahara et al.&nbsp;2020).For molecular details of cornified envelope formation during differentiation of stratum granulosum keratinocytes into corneocytes, please refer to Reactome pathway “Formation of the cornified envelope”, a subpathway of the Reactome “Keratinization” pathway (Blumenberg, M. 2016. “Keratinization.” Reactome - a Curated Knowledgebase of Biological Pathways 58 (September). https://doi.org/10.3180/r hsa 6805567.1).</p>
<div id="a680bc59" class="cell" data-execution_count="10">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Statistics of variable num_sentences:"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>print_stats(df[<span class="st">'num_sentences'</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Statistics of variable num_sentences:

Descriptive Statistics:
Mean: 3.61
Median: 3.00
Standard Deviation: 3.59
Minimum: 1
Maximum: 128
25th percentile (Q1): 2.00
75th percentile (Q3): 4.00
IQR (Interquartile Range): 2.00</code></pre>
</div>
</div>
<section id="repeated-sentences" class="level1">
<h1>Repeated sentences</h1>
<div id="cfa453f9" class="cell" data-execution_count="11">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>sentence_counts <span class="op">=</span> df_sentences[<span class="st">'sentence'</span>].value_counts().reset_index()</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>sentence_counts.columns <span class="op">=</span> [<span class="st">'sentence'</span>, <span class="st">'count'</span>]</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>repeated_sentences <span class="op">=</span> sentence_counts[sentence_counts[<span class="st">'count'</span>] <span class="op">&gt;</span> <span class="dv">1</span>]</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>repeated_sentences</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="26">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">sentence</th>
<th data-quarto-table-cell-role="th">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>This COVID-19 event has been created by a comb...</td>
<td>42</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>This reaction takes place in the 'nucleus'.</td>
<td>36</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Experiments using human cord blood CD4(+) T ce...</td>
<td>34</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Most early refolding studies were performed wi...</td>
<td>24</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Mutations within the C-propeptides further sug...</td>
<td>24</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">3142</td>
<td>In mammalian cells, LCMT1 knockdown results in...</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3143</td>
<td>FKBP1A dissociates from TGFBR1 after it forms ...</td>
<td>2</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">3144</td>
<td>However, BORA was recovered from the cytosolic...</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3145</td>
<td>GH38 members are found in the Golgi complex, l...</td>
<td>2</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">3146</td>
<td>The affinity of prednisolone (PREDL) to SERPIN...</td>
<td>2</td>
</tr>
</tbody>
</table>

<p>3147 rows × 2 columns</p>
</div>
</div>
</div>
<div id="a102e992" class="cell" data-execution_count="12">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>sentence_to_check <span class="op">=</span> repeated_sentences.iloc[<span class="dv">0</span>][<span class="st">'sentence'</span>]</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>notes_with_sentence <span class="op">=</span> df_sentences[df_sentences[<span class="st">'sentence'</span>] <span class="op">==</span> sentence_to_check]</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">4</span>):</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    note <span class="op">=</span> notes_with_sentence[<span class="st">'original_notes'</span>].iloc[i]</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    display(Markdown(<span class="ss">f"**Original notes at index </span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">:**  </span><span class="ch">\n</span><span class="sc">{</span>note<span class="sc">}</span><span class="ch">\n</span><span class="ss">"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display cell-output-markdown">
<p><strong>Original notes at index 0:</strong><br>
This COVID-19 event has been created by a combination of computational inference (see https://reactome.org/documentation/inferred-events) from SARS-CoV-1 data and manual curation, as described in the summation for the overall SARS-CoV-2 infection pathway.Nonstructural protein 15 (nsp15) of the SARS coronavirus (SARS-CoV-1) binds to the replication-transcription complex (RTC) through interaction with nsp8 (Imbert et al.&nbsp;2008). This interaction appears to be conserved in other coronaviruses, such as mouse hepatitis virus (MHV) (Athmer et al.&nbsp;2017). nsp15 is an endonuclease characteristic for the order Nidovirales that includes the family Coronaviridae. nsp15 preferentially cleaves 3’ of uridines, generating 2’-3’ cyclic phosphates after cleavage. nsp15 requires Mn2+ ions for catalytic activity. Functional nsp15 is needed for production of viable virions and for viral transcription (Guarino et al.&nbsp;2005, Ricagno et al.&nbsp;2006, Bhardwaj et al.&nbsp;2006, Joseph et al.&nbsp;2007, Bhardwaj et al.&nbsp;2008, Bhardwaj et al.&nbsp;2012). The biological role of nsp15 has not been elucidated. It may degrade host mRNAs to shut down host translation, but so far no human or viral RNA targets have been identified.</p>
</div>
<div class="cell-output cell-output-display cell-output-markdown">
<p><strong>Original notes at index 1:</strong><br>
This COVID-19 event has been created by a combination of computational inference (see https://reactome.org/documentation/inferred-events) from SARS-CoV-1 data and manual curation, as described in the summation for the overall SARS-CoV-2 infection pathway.The SARS-CoV nsp3 was shown to bind ORF7a and Nsp6 by using proteomics analysis (Neuman et al, 2008)</p>
</div>
<div class="cell-output cell-output-display cell-output-markdown">
<p><strong>Original notes at index 2:</strong><br>
This COVID-19 event has been created by a combination of computational inference (see https://reactome.org/documentation/inferred-events) from SARS-CoV-1 data and manual curation, as described in the summation for the overall SARS-CoV-2 infection pathway.Non-structural protein 14 (nsp14) of the human SARS coronavirus is a bifunctional enzyme bearing 3’-5 exoribonuclease activity involved in replication fidelity and RNA cap N7-guanine methyltransferase activity involved in 5’-RNA capping. nsp14 binds to the minimal replication and transcription complex (RTC), composed of nsp7, nsp8, and nsp12, by directly binding to nsp12 (the main RNA-dependent RNA polymerase). Binding of nsp14 does not affect the processivity of the RTC (Minskaia et al.&nbsp;2006, Subissi et al.&nbsp;2014).</p>
</div>
<div class="cell-output cell-output-display cell-output-markdown">
<p><strong>Original notes at index 3:</strong><br>
This COVID-19 event has been created by a combination of computational inference (see https://reactome.org/documentation/inferred-events) from SARS-CoV-1 data and manual curation, as described in the summation for the overall SARS-CoV-2 infection pathway.On the final assembly, proteomics data suggest that the NAB−βSM−TM1 domains of nsp3 can interact with nsp7 − 8, as well as nsps 12–16, and the domain Y1 plus CoV-Y interacts with nsp9 and nsp12 (Imbert et al., 2008). Also a PL2pro−NAB−βSM−TM1 construct of Nsp3 can bind Nsp4 and Nsp12, while the region from TM1 to the end of Nsp3 only binds Nsp8 (Pan et al., 2008).</p>
</div>
</div>
<!-- -->

</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb15" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Notes Analysis"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">    html:</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">        code-fold: true</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">        code-tools: true</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="an">jupyter:</span><span class="co"> python3</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy <span class="im">import</span> stats</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> descriptive_stat_functions <span class="im">import</span> plot_data, print_stats, extract_references</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> IPython.display <span class="im">import</span> display, Markdown</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>ggplot_colors <span class="op">=</span> plt.rcParams[<span class="st">'axes.prop_cycle'</span>].by_key()[<span class="st">'color'</span>]</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">"./reaction_sentences.parquet"</span>)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>df_sentences <span class="op">=</span> pd.read_parquet(<span class="st">"./sentences.parquet"</span>)</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>Reaction Sentences dataframe</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>df.head(<span class="dv">10</span>)</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of rows:</span><span class="sc">{</span><span class="bu">len</span>(df)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>Sentences dataframe</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>df_sentences.head(<span class="dv">10</span>)</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of rows:</span><span class="sc">{</span><span class="bu">len</span>(df_sentences)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: asis</span></span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a><span class="bu">max</span> <span class="op">=</span> <span class="dv">5</span></span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">0</span>,<span class="bu">max</span>):</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>  display(Markdown(<span class="ss">f"</span><span class="ch">\n</span><span class="ss"> **Original notes at index </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">:**  </span><span class="ch">\n</span><span class="sc">{</span>df[<span class="st">'original_notes'</span>]<span class="sc">.</span>iloc[idx]<span class="sc">}</span><span class="ss"> </span><span class="ch">\n</span><span class="ss">"</span>))</span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>  sentences <span class="op">=</span> <span class="st">"  </span><span class="ch">\n</span><span class="st">"</span>.join([<span class="ss">f"**-** </span><span class="sc">{</span>s<span class="sc">}</span><span class="ss">"</span> <span class="cf">for</span> s <span class="kw">in</span> df[<span class="st">'sentences'</span>].iloc[idx]])</span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>  display(Markdown(<span class="ss">f"**Sentences of note at index </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">:**  </span><span class="ch">\n</span><span class="sc">{</span>sentences<span class="sc">}</span><span class="ss">"</span>))</span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a>df.sort_values(by<span class="op">=</span><span class="st">"num_sentences"</span>, ascending<span class="op">=</span><span class="va">False</span>).head(<span class="dv">10</span>)</span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: asis</span></span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a>display(Markdown(<span class="ss">f"**Note with max value of sentences**</span><span class="ch">\n</span><span class="sc">{</span>df[<span class="st">'original_notes'</span>]<span class="sc">.</span>iloc[df[<span class="st">'num_sentences'</span>].idxmax()]<span class="sc">}</span><span class="ch">\n</span><span class="ss">"</span>))</span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-82"><a href="#cb15-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-83"><a href="#cb15-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-86"><a href="#cb15-86" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-87"><a href="#cb15-87" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Statistics of variable num_sentences:"</span>)</span>
<span id="cb15-88"><a href="#cb15-88" aria-hidden="true" tabindex="-1"></a>print_stats(df[<span class="st">'num_sentences'</span>])</span>
<span id="cb15-89"><a href="#cb15-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-90"><a href="#cb15-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-91"><a href="#cb15-91" aria-hidden="true" tabindex="-1"></a><span class="fu"># Repeated sentences </span></span>
<span id="cb15-92"><a href="#cb15-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-95"><a href="#cb15-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-96"><a href="#cb15-96" aria-hidden="true" tabindex="-1"></a>sentence_counts <span class="op">=</span> df_sentences[<span class="st">'sentence'</span>].value_counts().reset_index()</span>
<span id="cb15-97"><a href="#cb15-97" aria-hidden="true" tabindex="-1"></a>sentence_counts.columns <span class="op">=</span> [<span class="st">'sentence'</span>, <span class="st">'count'</span>]</span>
<span id="cb15-98"><a href="#cb15-98" aria-hidden="true" tabindex="-1"></a>repeated_sentences <span class="op">=</span> sentence_counts[sentence_counts[<span class="st">'count'</span>] <span class="op">&gt;</span> <span class="dv">1</span>]</span>
<span id="cb15-99"><a href="#cb15-99" aria-hidden="true" tabindex="-1"></a>repeated_sentences</span>
<span id="cb15-100"><a href="#cb15-100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-101"><a href="#cb15-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-104"><a href="#cb15-104" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-105"><a href="#cb15-105" aria-hidden="true" tabindex="-1"></a>sentence_to_check <span class="op">=</span> repeated_sentences.iloc[<span class="dv">0</span>][<span class="st">'sentence'</span>]</span>
<span id="cb15-106"><a href="#cb15-106" aria-hidden="true" tabindex="-1"></a>notes_with_sentence <span class="op">=</span> df_sentences[df_sentences[<span class="st">'sentence'</span>] <span class="op">==</span> sentence_to_check]</span>
<span id="cb15-107"><a href="#cb15-107" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">4</span>):</span>
<span id="cb15-108"><a href="#cb15-108" aria-hidden="true" tabindex="-1"></a>    note <span class="op">=</span> notes_with_sentence[<span class="st">'original_notes'</span>].iloc[i]</span>
<span id="cb15-109"><a href="#cb15-109" aria-hidden="true" tabindex="-1"></a>    display(Markdown(<span class="ss">f"**Original notes at index </span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">:**  </span><span class="ch">\n</span><span class="sc">{</span>note<span class="sc">}</span><span class="ch">\n</span><span class="ss">"</span>))</span>
<span id="cb15-110"><a href="#cb15-110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-111"><a href="#cb15-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-114"><a href="#cb15-114" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb15-115"><a href="#cb15-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>