<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.29">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Pathways Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="pathways_files/libs/clipboard/clipboard.min.js"></script>
<script src="pathways_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="pathways_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="pathways_files/libs/quarto-html/popper.min.js"></script>
<script src="pathways_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="pathways_files/libs/quarto-html/anchor.min.js"></script>
<link href="pathways_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="pathways_files/libs/quarto-html/quarto-syntax-highlighting-0815c480559380816a4d1ea211a47e91.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="pathways_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="pathways_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="pathways_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Pathways Analysis</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>How many unique vales among the variables?</p>
<div id="f35c4102" class="cell" data-execution_count="2">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the big dataset </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">"./reactions.parquet"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>df.nunique()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="26">
<pre><code>file_id             2769
reaction_id        15357
original_notes     14802
notes              14799
only_references    10458
snippet            70334
dtype: int64</code></pre>
</div>
</div>
<section id="reactions-per-pathways" class="level1">
<h1>Reactions per pathways</h1>
<section id="raw-data" class="level2">
<h2 class="anchored" data-anchor-id="raw-data">Raw data</h2>
<div id="d2686a2f" class="cell" data-execution_count="3">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read Reaction Count data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>df_count <span class="op">=</span> pd.read_parquet(<span class="st">"./reaction_counts.parquet"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>reaction_count <span class="op">=</span> df_count[<span class="st">'reaction_count'</span>]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary statistics</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>plot_data(reaction_count, boxplot <span class="op">=</span> <span class="va">True</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>print_stats(reaction_count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-4-output-1.png" width="558" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Descriptive Statistics:
Mean: 25.61
Median: 7.00
Standard Deviation: 108.05
Minimum: 1
Maximum: 2582
25th percentile (Q1): 3.00
75th percentile (Q3): 18.00
IQR (Interquartile Range): 15.00</code></pre>
</div>
</div>
<div id="8eee3795" class="cell" data-execution_count="4">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the row with the maximum reaction count</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>max_idx <span class="op">=</span> df_count[<span class="st">'reaction_count'</span>].idxmax()  </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>max_row <span class="op">=</span> df_count.loc[max_idx]   </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">File with Maximum Reaction Count: </span><span class="sc">{</span>max_row[<span class="st">'file_id'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"reaction_count: </span><span class="sc">{</span>max_row[<span class="st">'reaction_count'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#R-HSA-162582.sbml</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort the DataFrame by reaction_count in descending order</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>sorted_df <span class="op">=</span> df_count.sort_values(by<span class="op">=</span><span class="st">'reaction_count'</span>, ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the top files</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">Files sorted by reaction_count (descending):"</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>sorted_df.head(<span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
File with Maximum Reaction Count: R-HSA-162582.sbml
reaction_count: 2582

Files sorted by reaction_count (descending):</code></pre>
</div>
<div class="cell-output cell-output-display" data-execution_count="28">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">file_id</th>
<th data-quarto-table-cell-role="th">reaction_count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">2768</td>
<td>R-HSA-162582.sbml</td>
<td>2582</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2767</td>
<td>R-HSA-1430728.sbml</td>
<td>2355</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2761</td>
<td>R-HSA-1643685.sbml</td>
<td>1940</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2765</td>
<td>R-HSA-168256.sbml</td>
<td>1737</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2227</td>
<td>R-HSA-74160.sbml</td>
<td>1141</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2093</td>
<td>R-HSA-5663205.sbml</td>
<td>999</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2313</td>
<td>R-HSA-556833.sbml</td>
<td>994</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1608</td>
<td>R-HSA-1266738.sbml</td>
<td>956</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2760</td>
<td>R-HSA-73857.sbml</td>
<td>940</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2759</td>
<td>R-HSA-392499.sbml</td>
<td>891</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">1770</td>
<td>R-HSA-212436.sbml</td>
<td>879</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1219</td>
<td>R-HSA-1280215.sbml</td>
<td>791</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2763</td>
<td>R-HSA-9006934.sbml</td>
<td>759</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2766</td>
<td>R-HSA-9824446.sbml</td>
<td>753</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2077</td>
<td>R-HSA-168249.sbml</td>
<td>744</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="log-scaled-data" class="level2">
<h2 class="anchored" data-anchor-id="log-scaled-data">Log-scaled data</h2>
<p>Log10 transformation of the data</p>
<div id="bfc2ff8b" class="cell" data-execution_count="5">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-scaled data ---------------------------------</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>plot_data(reaction_count, </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>          logscale_x <span class="op">=</span> <span class="va">True</span>, </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig2.png"</span>, </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>, </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>          xlabel<span class="op">=</span> <span class="st">"Log10 Reactions"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-6-output-1.png" width="574" height="374" class="figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="filtered-data-90th-percentile" class="level2">
<h2 class="anchored" data-anchor-id="filtered-data-90th-percentile">Filtered data 90th percentile</h2>
<div id="427e17b5" class="cell" data-execution_count="6">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>perc <span class="op">=</span> np.percentile(df_count[<span class="st">'reaction_count'</span>], <span class="dv">90</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>df_filtered <span class="op">=</span> df_count[df_count[<span class="st">'reaction_count'</span>] <span class="op">&lt;=</span> perc]</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>reaction_count_filtered <span class="op">=</span> df_filtered[<span class="st">'reaction_count'</span>]</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>plot_data(reaction_count_filtered, boxplot <span class="op">=</span> <span class="va">True</span>, outputname <span class="op">=</span> <span class="st">"fig3.png"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>print_stats(reaction_count_filtered)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-7-output-1.png" width="558" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Descriptive Statistics:
Mean: 9.29
Median: 6.00
Standard Deviation: 9.60
Minimum: 1
Maximum: 43
25th percentile (Q1): 2.00
75th percentile (Q3): 13.00
IQR (Interquartile Range): 11.00</code></pre>
</div>
</div>
</section>
</section>
<section id="references-per-pathway-distribution" class="level1">
<h1>References per Pathway distribution</h1>
<div id="724c37ee" class="cell" data-execution_count="7">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example: </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">'''(Duncan et al. 2002, Sundheim et al. 2006, Chen et al. 2010, Dango et al. 2011);</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"> (Duncan et al. 2002, Sundheim et al. 2006)'''</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply extraction</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'parsed_references'</span>] <span class="op">=</span> df[<span class="st">'only_references'</span>].<span class="bu">apply</span>(extract_references)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># How many unique references in general</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># How many references per file </span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># List of all ref</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>refs_list_per_file <span class="op">=</span> df.groupby(<span class="st">'file_id'</span>)[<span class="st">'parsed_references'</span>].<span class="bu">apply</span>(</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> lists: [ref <span class="cf">for</span> refs <span class="kw">in</span> lists <span class="cf">for</span> ref <span class="kw">in</span> refs]</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>refs_per_file <span class="op">=</span> refs_list_per_file.<span class="bu">apply</span>(<span class="kw">lambda</span> refs: <span class="bu">len</span>(refs))</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># How many unique references per file </span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>unique_refs_per_file <span class="op">=</span> refs_list_per_file.<span class="bu">apply</span>(<span class="kw">lambda</span> refs: <span class="bu">len</span>(<span class="bu">set</span>(refs)))</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>df_refs_per_file <span class="op">=</span> pd.DataFrame({</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">'file_id'</span>: refs_list_per_file.index,</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">'all_references'</span>: refs_list_per_file.values,</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">'total_references'</span>: refs_per_file.values,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">'unique_references'</span>: unique_refs_per_file.values</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>df_refs_per_file.head(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="31">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">file_id</th>
<th data-quarto-table-cell-role="th">all_references</th>
<th data-quarto-table-cell-role="th">total_references</th>
<th data-quarto-table-cell-role="th">unique_references</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>R-HSA-1059683.sbml</td>
<td>[Schmitz et al. 2000, Lehmann et al. 2003, Hil...</td>
<td>89</td>
<td>74</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>R-HSA-109581.sbml</td>
<td>[Llambi et al. 2001, Morishima et al., 1999, B...</td>
<td>484</td>
<td>245</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>R-HSA-109582.sbml</td>
<td>[Reid et al. 2011, Kurosawa et al. 2000, Campb...</td>
<td>1445</td>
<td>834</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>R-HSA-109606.sbml</td>
<td>[Hershko and Ginsberg 2004, Hershko and Ginsbe...</td>
<td>260</td>
<td>154</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>R-HSA-109703.sbml</td>
<td>[Marganiello et al, 1995, Degerman et al, 1997...</td>
<td>16</td>
<td>14</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>R-HSA-109704.sbml</td>
<td>[Maira et al. 2001, Du et al. 2003, Marganiell...</td>
<td>29</td>
<td>26</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>R-HSA-110056.sbml</td>
<td>[Stahl et al. 1995, Kim et al. 1998, Fukada et...</td>
<td>8</td>
<td>8</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>R-HSA-110312.sbml</td>
<td>[Nelson et al. 1996, Hara et al. 2010, Kikuchi...</td>
<td>21</td>
<td>16</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>R-HSA-110313.sbml</td>
<td>[Nelson et al. 1996, Hara et al. 2010, Kikuchi...</td>
<td>99</td>
<td>62</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>R-HSA-110314.sbml</td>
<td>[Hoege et al. 2002, Park et al. 2014, Friedber...</td>
<td>14</td>
<td>9</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="9081c9eb" class="cell" data-execution_count="8">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">First reference:</span><span class="ch">\n</span><span class="sc">{</span>df_refs_per_file[<span class="st">'all_references'</span>]<span class="sc">.</span>iloc[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Total number of references in all files (with duplicates): </span><span class="sc">{</span><span class="bu">sum</span>(df_refs_per_file.total_references.values)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Total number of unique references across all files: </span><span class="sc">{</span><span class="bu">sum</span>(df_refs_per_file.unique_references.values)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span> df_refs_per_file[<span class="st">'total_references'</span>],</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig4.png"</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"References per file (with duplicates)"</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">1</span>],</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'total_references'</span>],</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig5.png"</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file with log scale x (with duplicates)"</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"log(References)"</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">1</span>],</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>          logscale_x <span class="op">=</span> <span class="va">True</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'total_references'</span>],</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig6.png"</span>,</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file cutted (with duplicates)"</span>,</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">1</span>],</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>          cut_percentile<span class="op">=</span>(<span class="dv">0</span>,<span class="dv">90</span>)</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>print_stats(df_refs_per_file[<span class="st">'total_references'</span>])</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span> df_refs_per_file[<span class="st">'unique_references'</span>],</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig7.png"</span>,</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"References per file (without duplicates)"</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">2</span>],</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'unique_references'</span>],</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig8.png"</span>,</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file with log scale x (without duplicates)"</span>,</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"log(References)"</span>,</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">2</span>],</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>          logscale_x <span class="op">=</span> <span class="va">True</span></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'unique_references'</span>],</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig9.png"</span>,</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file cutted (without duplicates)"</span>,</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">2</span>],</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>          cut_percentile<span class="op">=</span>(<span class="dv">0</span>,<span class="dv">90</span>)</span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>          )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
First reference:
['Schmitz et al. 2000', 'Lehmann et al. 2003', 'Hilton et al. 1998', 'Narazaki et al. 1998', 'Yasukawa et al. 1999', 'Nicholson et al. 1999', 'Greenhalgh and Hilton 2001', 'Croker et al. 2003', 'Rose-John et al. 2006', 'Jostock et al. 2001', 'Gerhartz et al. 1996', 'Hemmann et al. 1996', 'Wen and Darnell 1997', 'Shen et al. 2004', 'Jain et al. 1999', 'Jain et al. 1999', 'Abe et al. 2001', 'Chung et al. 1997', 'Decker and Kovarik 2000', 'Reich 2009', 'Scheller and Rose-John 2006', 'Grotzinger et al. 1999', 'Boulanger et al. 2003', 'Boulanger et al. 2003', 'Skiniotis et al. 2005', 'Suthaus et al. 2010', 'Riech and Liu 2006', 'Schaper et al. 1998', 'Lehmann et al. 2003', 'Fischer', '2004', 'Schaper et al. 1998', 'Fukada et al. 1996', 'Kim and Baumann 1999', 'Dance et al. 2008', 'Takahashi-Tezuka et al. 1998', 'Eulenfeld and Schaper 2009', 'Zhang et al. 1995', 'Reich and Liu 2006', 'Wang and Fuller 1994', 'Reich and Liu 2006', 'Stahl et al. 1995', 'Gerhartz et al. 1996', 'Schaper et al. 1998', 'Lehmann et al. 2003', 'Fischer', '2004', 'Schaper et al', '1998', 'Lu et al. 2001', 'Symes et al. 1997', 'Kim and Baumann', '1999', 'Tanaka et al. 2008', 'Stahl et al. 1994', 'Narazaki et al. 1994', 'Haan et al. 2000', 'Heinrich et al. 2003', 'Haan et al. 2001', 'Hilkens et al. 2001', 'Guschin et al. 1995', 'Guschin et al. 1995', 'Rodig et al. 1998', 'Haan et al. 2008', 'Lindauer et al. 2001', 'Liu et al. 1997', 'Velazquez et al. 1995', 'Oakes et al. 1996', 'Haan et al. 2002', 'Rose-John et al. 2006', 'Akira et al. 1994', 'Zhong et al. 1994', 'Hemmann et al. 1996', 'Kaptein et al. 1996', 'Shuai et al. 1994', 'Zhong et al. 1994', 'Kaur et al. 2020', 'Choy et al. 2002', 'Heo 2020', 'Rosso et al. 2020', 'Duchow and Bellmann-Strobl 2021', 'Xu et al. 2020', 'Strand et al. 2020', 'Xu et al. 2020', 'Sarhan et al. 2021', 'Khan et al. 2021', 'Hornby et al. 2021', 'Luo et al. 2020', 'Zhang et al. 2020a']

Total number of references in all files (with duplicates): 330135
Total number of unique references across all files: 147741</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-9-output-2.png" width="566" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/usr/local/lib/python3.10/dist-packages/pandas/core/arraylike.py:399: RuntimeWarning: divide by zero encountered in log10
  result = getattr(ufunc, method)(*inputs, **kwargs)</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-9-output-4.png" width="576" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-9-output-5.png" width="567" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Descriptive Statistics:
Mean: 119.23
Median: 20.00
Standard Deviation: 547.74
Minimum: 0
Maximum: 15558
25th percentile (Q1): 6.00
75th percentile (Q3): 71.00
IQR (Interquartile Range): 65.00</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-9-output-7.png" width="567" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/usr/local/lib/python3.10/dist-packages/pandas/core/arraylike.py:399: RuntimeWarning: divide by zero encountered in log10
  result = getattr(ufunc, method)(*inputs, **kwargs)</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-9-output-9.png" width="592" height="374" class="figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="pathways_files/figure-html/cell-9-output-10.png" width="567" height="374" class="figure-img"></p>
</figure>
</div>
</div>
</div>
<section id="filtered-data-without-any-reference" class="level2">
<h2 class="anchored" data-anchor-id="filtered-data-without-any-reference">Filtered data without any reference</h2>
<div id="b57f2ecd" class="cell" data-execution_count="9">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># No references data -----------------------</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>df_nocit <span class="op">=</span> df[df[<span class="st">'only_references'</span>].isna()]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic info</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Dataset length:"</span>, <span class="bu">len</span>(df_nocit))</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Columns in dataset:"</span>, df_nocit.columns.tolist())</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co"># First rows </span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">First 5 rows:"</span>)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>df_nocit.head(<span class="dv">5</span>)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Display a sample of the dataset.</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>num_samples <span class="op">=</span> <span class="dv">5</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over the first num_samples rows and print full content without extra spaces.</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">--- Displaying full content for sample rows ---"</span>)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx <span class="kw">in</span> df_nocit.index[:num_samples]:</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">--- Full content for sample row at index </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss"> ---"</span>)</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    <span class="co">#print(f"Snippet:\n{df_nocit.at[idx, 'snippet']}")</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Original Notes:</span><span class="ch">\n</span><span class="sc">{</span>df_nocit<span class="sc">.</span>at[idx, <span class="st">'original_notes'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Cleaned Notes (without citations):</span><span class="ch">\n</span><span class="sc">{</span>df_nocit<span class="sc">.</span>at[idx, <span class="st">'notes'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Extracted Citations:</span><span class="ch">\n</span><span class="sc">{</span>df_nocit<span class="sc">.</span>at[idx, <span class="st">'only_references'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"-"</span> <span class="op">*</span> <span class="dv">40</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Dataset length: 13309
Columns in dataset: ['file_id', 'reaction_id', 'original_notes', 'notes', 'only_references', 'snippet', 'parsed_references']

First 5 rows:

--- Displaying full content for sample rows ---

--- Full content for sample row at index 18 ---

Original Notes:
The Calreticulin gene is transcribed to yield mRNA and the mRNA is translated to yield protein.

Cleaned Notes (without citations):
The Calreticulin gene is transcribed to yield mRNA and the mRNA is translated to yield protein.

Extracted Citations:
None
----------------------------------------

--- Full content for sample row at index 19 ---

Original Notes:
The Endoplasmin (HSP90B1) gene is transcribed to yield mRNA and the mRNA is translated to yield protein.

Cleaned Notes (without citations):
The Endoplasmin (HSP90B1) gene is transcribed to yield mRNA and the mRNA is translated to yield protein.

Extracted Citations:
None
----------------------------------------

--- Full content for sample row at index 20 ---

Original Notes:
The BIP (HSPA5) gene is transcribed to yield mRNA and the mRNA is translated to yield protein.

Cleaned Notes (without citations):
The BIP (HSPA5) gene is transcribed to yield mRNA and the mRNA is translated to yield protein.

Extracted Citations:
None
----------------------------------------

--- Full content for sample row at index 21 ---

Original Notes:
The XBP1 gene is transcribed to yield mRNA, the mRNA is spliced to yield Xbp(s) mRNA, and the mRNA is translated to yield protein. The N-terminal fragment of ATF6-alpha contains a bZIP domain and binds the sequence CCACG in ER Stress Response Elements (ERSEs). ATF6-alpha binds ERSEs together with the heterotrimeric transcription factor NF-Y, which binds the sequence CCAAT in the ERSEs, and together the two factors activate transcription of ER stress-responsive genes. Evidence from overexpression and knockdowns indicates that ATF6-alpha is a potent activator but its homolog ATF6-beta is not and ATF6-beta may actually reduce expression of ER stress proteins.

Cleaned Notes (without citations):
The XBP1 gene is transcribed to yield mRNA, the mRNA is spliced to yield Xbp(s) mRNA, and the mRNA is translated to yield protein. The N-terminal fragment of ATF6-alpha contains a bZIP domain and binds the sequence CCACG in ER Stress Response Elements (ERSEs). ATF6-alpha binds ERSEs together with the heterotrimeric transcription factor NF-Y, which binds the sequence CCAAT in the ERSEs, and together the two factors activate transcription of ER stress-responsive genes. Evidence from overexpression and knockdowns indicates that ATF6-alpha is a potent activator but its homolog ATF6-beta is not and ATF6-beta may actually reduce expression of ER stress proteins.

Extracted Citations:
None
----------------------------------------

--- Full content for sample row at index 26 ---

Original Notes:
XRCC4 and DNA ligase 4 are recruited to the complex containing viral DNA.

Cleaned Notes (without citations):
XRCC4 and DNA ligase 4 are recruited to the complex containing viral DNA.

Extracted Citations:
None
----------------------------------------</code></pre>
</div>
</div>
<!-- -->

</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb18" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Pathways Analysis"</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">    html:</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">        code-fold: true</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">        code-tools: true</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="an">jupyter:</span><span class="co"> python3</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy <span class="im">import</span> stats</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> descriptive_stat_functions <span class="im">import</span> plot_data, print_stats, extract_references</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>ggplot_colors <span class="op">=</span> plt.rcParams[<span class="st">'axes.prop_cycle'</span>].by_key()[<span class="st">'color'</span>]</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>How many unique vales among the variables? </span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the big dataset </span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">"./reactions.parquet"</span>)</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>df.nunique()</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a><span class="fu"># Reactions per pathways</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a><span class="fu">## Raw data</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Read Reaction Count data</span></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>df_count <span class="op">=</span> pd.read_parquet(<span class="st">"./reaction_counts.parquet"</span>)</span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>reaction_count <span class="op">=</span> df_count[<span class="st">'reaction_count'</span>]</span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary statistics</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>plot_data(reaction_count, boxplot <span class="op">=</span> <span class="va">True</span>)</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>print_stats(reaction_count)</span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the row with the maximum reaction count</span></span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>max_idx <span class="op">=</span> df_count[<span class="st">'reaction_count'</span>].idxmax()  </span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>max_row <span class="op">=</span> df_count.loc[max_idx]   </span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">File with Maximum Reaction Count: </span><span class="sc">{</span>max_row[<span class="st">'file_id'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"reaction_count: </span><span class="sc">{</span>max_row[<span class="st">'reaction_count'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a><span class="co">#R-HSA-162582.sbml</span></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort the DataFrame by reaction_count in descending order</span></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a>sorted_df <span class="op">=</span> df_count.sort_values(by<span class="op">=</span><span class="st">'reaction_count'</span>, ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the top files</span></span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">Files sorted by reaction_count (descending):"</span>)</span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>sorted_df.head(<span class="dv">15</span>)</span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a><span class="fu">## Log-scaled data</span></span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a>Log10 transformation of the data</span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-scaled data ---------------------------------</span></span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a>plot_data(reaction_count, </span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a>          logscale_x <span class="op">=</span> <span class="va">True</span>, </span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig2.png"</span>, </span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>, </span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a>          xlabel<span class="op">=</span> <span class="st">"Log10 Reactions"</span>)</span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a><span class="fu">## Filtered data 90th percentile </span></span>
<span id="cb18-82"><a href="#cb18-82" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-83"><a href="#cb18-83" aria-hidden="true" tabindex="-1"></a>perc <span class="op">=</span> np.percentile(df_count[<span class="st">'reaction_count'</span>], <span class="dv">90</span>)</span>
<span id="cb18-84"><a href="#cb18-84" aria-hidden="true" tabindex="-1"></a>df_filtered <span class="op">=</span> df_count[df_count[<span class="st">'reaction_count'</span>] <span class="op">&lt;=</span> perc]</span>
<span id="cb18-85"><a href="#cb18-85" aria-hidden="true" tabindex="-1"></a>reaction_count_filtered <span class="op">=</span> df_filtered[<span class="st">'reaction_count'</span>]</span>
<span id="cb18-86"><a href="#cb18-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-87"><a href="#cb18-87" aria-hidden="true" tabindex="-1"></a>plot_data(reaction_count_filtered, boxplot <span class="op">=</span> <span class="va">True</span>, outputname <span class="op">=</span> <span class="st">"fig3.png"</span>)</span>
<span id="cb18-88"><a href="#cb18-88" aria-hidden="true" tabindex="-1"></a>print_stats(reaction_count_filtered)</span>
<span id="cb18-89"><a href="#cb18-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-90"><a href="#cb18-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-91"><a href="#cb18-91" aria-hidden="true" tabindex="-1"></a><span class="fu"># References per Pathway distribution</span></span>
<span id="cb18-94"><a href="#cb18-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-95"><a href="#cb18-95" aria-hidden="true" tabindex="-1"></a><span class="co"># Example: </span></span>
<span id="cb18-96"><a href="#cb18-96" aria-hidden="true" tabindex="-1"></a><span class="co">'''(Duncan et al. 2002, Sundheim et al. 2006, Chen et al. 2010, Dango et al. 2011);</span></span>
<span id="cb18-97"><a href="#cb18-97" aria-hidden="true" tabindex="-1"></a><span class="co"> (Duncan et al. 2002, Sundheim et al. 2006)'''</span></span>
<span id="cb18-98"><a href="#cb18-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-99"><a href="#cb18-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply extraction</span></span>
<span id="cb18-100"><a href="#cb18-100" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'parsed_references'</span>] <span class="op">=</span> df[<span class="st">'only_references'</span>].<span class="bu">apply</span>(extract_references)</span>
<span id="cb18-101"><a href="#cb18-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-102"><a href="#cb18-102" aria-hidden="true" tabindex="-1"></a><span class="co"># How many unique references in general</span></span>
<span id="cb18-103"><a href="#cb18-103" aria-hidden="true" tabindex="-1"></a><span class="co"># How many references per file </span></span>
<span id="cb18-104"><a href="#cb18-104" aria-hidden="true" tabindex="-1"></a><span class="co"># List of all ref</span></span>
<span id="cb18-105"><a href="#cb18-105" aria-hidden="true" tabindex="-1"></a>refs_list_per_file <span class="op">=</span> df.groupby(<span class="st">'file_id'</span>)[<span class="st">'parsed_references'</span>].<span class="bu">apply</span>(</span>
<span id="cb18-106"><a href="#cb18-106" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> lists: [ref <span class="cf">for</span> refs <span class="kw">in</span> lists <span class="cf">for</span> ref <span class="kw">in</span> refs]</span>
<span id="cb18-107"><a href="#cb18-107" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-108"><a href="#cb18-108" aria-hidden="true" tabindex="-1"></a>refs_per_file <span class="op">=</span> refs_list_per_file.<span class="bu">apply</span>(<span class="kw">lambda</span> refs: <span class="bu">len</span>(refs))</span>
<span id="cb18-109"><a href="#cb18-109" aria-hidden="true" tabindex="-1"></a><span class="co"># How many unique references per file </span></span>
<span id="cb18-110"><a href="#cb18-110" aria-hidden="true" tabindex="-1"></a>unique_refs_per_file <span class="op">=</span> refs_list_per_file.<span class="bu">apply</span>(<span class="kw">lambda</span> refs: <span class="bu">len</span>(<span class="bu">set</span>(refs)))</span>
<span id="cb18-111"><a href="#cb18-111" aria-hidden="true" tabindex="-1"></a>df_refs_per_file <span class="op">=</span> pd.DataFrame({</span>
<span id="cb18-112"><a href="#cb18-112" aria-hidden="true" tabindex="-1"></a>    <span class="st">'file_id'</span>: refs_list_per_file.index,</span>
<span id="cb18-113"><a href="#cb18-113" aria-hidden="true" tabindex="-1"></a>    <span class="st">'all_references'</span>: refs_list_per_file.values,</span>
<span id="cb18-114"><a href="#cb18-114" aria-hidden="true" tabindex="-1"></a>    <span class="st">'total_references'</span>: refs_per_file.values,</span>
<span id="cb18-115"><a href="#cb18-115" aria-hidden="true" tabindex="-1"></a>    <span class="st">'unique_references'</span>: unique_refs_per_file.values</span>
<span id="cb18-116"><a href="#cb18-116" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb18-117"><a href="#cb18-117" aria-hidden="true" tabindex="-1"></a>df_refs_per_file.head(<span class="dv">10</span>)</span>
<span id="cb18-118"><a href="#cb18-118" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-121"><a href="#cb18-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-122"><a href="#cb18-122" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">First reference:</span><span class="ch">\n</span><span class="sc">{</span>df_refs_per_file[<span class="st">'all_references'</span>]<span class="sc">.</span>iloc[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-123"><a href="#cb18-123" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Total number of references in all files (with duplicates): </span><span class="sc">{</span><span class="bu">sum</span>(df_refs_per_file.total_references.values)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-124"><a href="#cb18-124" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Total number of unique references across all files: </span><span class="sc">{</span><span class="bu">sum</span>(df_refs_per_file.unique_references.values)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-125"><a href="#cb18-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-126"><a href="#cb18-126" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span> df_refs_per_file[<span class="st">'total_references'</span>],</span>
<span id="cb18-127"><a href="#cb18-127" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig4.png"</span>,</span>
<span id="cb18-128"><a href="#cb18-128" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"References per file (with duplicates)"</span>,</span>
<span id="cb18-129"><a href="#cb18-129" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb18-130"><a href="#cb18-130" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">1</span>],</span>
<span id="cb18-131"><a href="#cb18-131" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span></span>
<span id="cb18-132"><a href="#cb18-132" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb18-133"><a href="#cb18-133" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'total_references'</span>],</span>
<span id="cb18-134"><a href="#cb18-134" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig5.png"</span>,</span>
<span id="cb18-135"><a href="#cb18-135" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file with log scale x (with duplicates)"</span>,</span>
<span id="cb18-136"><a href="#cb18-136" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"log(References)"</span>,</span>
<span id="cb18-137"><a href="#cb18-137" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">1</span>],</span>
<span id="cb18-138"><a href="#cb18-138" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb18-139"><a href="#cb18-139" aria-hidden="true" tabindex="-1"></a>          logscale_x <span class="op">=</span> <span class="va">True</span></span>
<span id="cb18-140"><a href="#cb18-140" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb18-141"><a href="#cb18-141" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'total_references'</span>],</span>
<span id="cb18-142"><a href="#cb18-142" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig6.png"</span>,</span>
<span id="cb18-143"><a href="#cb18-143" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file cutted (with duplicates)"</span>,</span>
<span id="cb18-144"><a href="#cb18-144" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb18-145"><a href="#cb18-145" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">1</span>],</span>
<span id="cb18-146"><a href="#cb18-146" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb18-147"><a href="#cb18-147" aria-hidden="true" tabindex="-1"></a>          cut_percentile<span class="op">=</span>(<span class="dv">0</span>,<span class="dv">90</span>)</span>
<span id="cb18-148"><a href="#cb18-148" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb18-149"><a href="#cb18-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-150"><a href="#cb18-150" aria-hidden="true" tabindex="-1"></a>print_stats(df_refs_per_file[<span class="st">'total_references'</span>])</span>
<span id="cb18-151"><a href="#cb18-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-152"><a href="#cb18-152" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span> df_refs_per_file[<span class="st">'unique_references'</span>],</span>
<span id="cb18-153"><a href="#cb18-153" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig7.png"</span>,</span>
<span id="cb18-154"><a href="#cb18-154" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"References per file (without duplicates)"</span>,</span>
<span id="cb18-155"><a href="#cb18-155" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb18-156"><a href="#cb18-156" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">2</span>],</span>
<span id="cb18-157"><a href="#cb18-157" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span></span>
<span id="cb18-158"><a href="#cb18-158" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb18-159"><a href="#cb18-159" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'unique_references'</span>],</span>
<span id="cb18-160"><a href="#cb18-160" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig8.png"</span>,</span>
<span id="cb18-161"><a href="#cb18-161" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file with log scale x (without duplicates)"</span>,</span>
<span id="cb18-162"><a href="#cb18-162" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"log(References)"</span>,</span>
<span id="cb18-163"><a href="#cb18-163" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">2</span>],</span>
<span id="cb18-164"><a href="#cb18-164" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb18-165"><a href="#cb18-165" aria-hidden="true" tabindex="-1"></a>          logscale_x <span class="op">=</span> <span class="va">True</span></span>
<span id="cb18-166"><a href="#cb18-166" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb18-167"><a href="#cb18-167" aria-hidden="true" tabindex="-1"></a>plot_data(data <span class="op">=</span>  df_refs_per_file[<span class="st">'unique_references'</span>],</span>
<span id="cb18-168"><a href="#cb18-168" aria-hidden="true" tabindex="-1"></a>          outputname <span class="op">=</span> <span class="st">"fig9.png"</span>,</span>
<span id="cb18-169"><a href="#cb18-169" aria-hidden="true" tabindex="-1"></a>          title <span class="op">=</span> <span class="st">"Referencese per file cutted (without duplicates)"</span>,</span>
<span id="cb18-170"><a href="#cb18-170" aria-hidden="true" tabindex="-1"></a>          xlabel <span class="op">=</span> <span class="st">"References"</span>,</span>
<span id="cb18-171"><a href="#cb18-171" aria-hidden="true" tabindex="-1"></a>          color <span class="op">=</span> ggplot_colors[<span class="dv">2</span>],</span>
<span id="cb18-172"><a href="#cb18-172" aria-hidden="true" tabindex="-1"></a>          boxplot <span class="op">=</span> <span class="va">True</span>,</span>
<span id="cb18-173"><a href="#cb18-173" aria-hidden="true" tabindex="-1"></a>          cut_percentile<span class="op">=</span>(<span class="dv">0</span>,<span class="dv">90</span>)</span>
<span id="cb18-174"><a href="#cb18-174" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb18-175"><a href="#cb18-175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-176"><a href="#cb18-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-177"><a href="#cb18-177" aria-hidden="true" tabindex="-1"></a><span class="fu">## Filtered data without any reference</span></span>
<span id="cb18-180"><a href="#cb18-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-181"><a href="#cb18-181" aria-hidden="true" tabindex="-1"></a><span class="co"># No references data -----------------------</span></span>
<span id="cb18-182"><a href="#cb18-182" aria-hidden="true" tabindex="-1"></a>df_nocit <span class="op">=</span> df[df[<span class="st">'only_references'</span>].isna()]</span>
<span id="cb18-183"><a href="#cb18-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-184"><a href="#cb18-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic info</span></span>
<span id="cb18-185"><a href="#cb18-185" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Dataset length:"</span>, <span class="bu">len</span>(df_nocit))</span>
<span id="cb18-186"><a href="#cb18-186" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Columns in dataset:"</span>, df_nocit.columns.tolist())</span>
<span id="cb18-187"><a href="#cb18-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-188"><a href="#cb18-188" aria-hidden="true" tabindex="-1"></a><span class="co"># First rows </span></span>
<span id="cb18-189"><a href="#cb18-189" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">First 5 rows:"</span>)</span>
<span id="cb18-190"><a href="#cb18-190" aria-hidden="true" tabindex="-1"></a>df_nocit.head(<span class="dv">5</span>)</span>
<span id="cb18-191"><a href="#cb18-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-192"><a href="#cb18-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Display a sample of the dataset.</span></span>
<span id="cb18-193"><a href="#cb18-193" aria-hidden="true" tabindex="-1"></a>num_samples <span class="op">=</span> <span class="dv">5</span></span>
<span id="cb18-194"><a href="#cb18-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-195"><a href="#cb18-195" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over the first num_samples rows and print full content without extra spaces.</span></span>
<span id="cb18-196"><a href="#cb18-196" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">--- Displaying full content for sample rows ---"</span>)</span>
<span id="cb18-197"><a href="#cb18-197" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx <span class="kw">in</span> df_nocit.index[:num_samples]:</span>
<span id="cb18-198"><a href="#cb18-198" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">--- Full content for sample row at index </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss"> ---"</span>)</span>
<span id="cb18-199"><a href="#cb18-199" aria-hidden="true" tabindex="-1"></a>    <span class="co">#print(f"Snippet:\n{df_nocit.at[idx, 'snippet']}")</span></span>
<span id="cb18-200"><a href="#cb18-200" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Original Notes:</span><span class="ch">\n</span><span class="sc">{</span>df_nocit<span class="sc">.</span>at[idx, <span class="st">'original_notes'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-201"><a href="#cb18-201" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Cleaned Notes (without citations):</span><span class="ch">\n</span><span class="sc">{</span>df_nocit<span class="sc">.</span>at[idx, <span class="st">'notes'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-202"><a href="#cb18-202" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Extracted Citations:</span><span class="ch">\n</span><span class="sc">{</span>df_nocit<span class="sc">.</span>at[idx, <span class="st">'only_references'</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-203"><a href="#cb18-203" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"-"</span> <span class="op">*</span> <span class="dv">40</span>)</span>
<span id="cb18-204"><a href="#cb18-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-205"><a href="#cb18-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-208"><a href="#cb18-208" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-209"><a href="#cb18-209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-210"><a href="#cb18-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-213"><a href="#cb18-213" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb18-214"><a href="#cb18-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>